freenet/src/main/java/ch/cyberduck/core/freenet/FreenetSession.java

```diff
@@ -47,6 +53,17 @@ protected DAVClient connect(final Proxy proxy, final HostKeyCallback key, final
         final HttpClientBuilder configuration = builder.build(proxy, this, prompt);
         configuration.setRedirectStrategy(new DAVRedirectStrategy(new PreferencesRedirectCallback()));
         configuration.setUserAgent(new FreenetUserAgentProvider().get());
+        configuration.addInterceptorLast(new HttpRequestInterceptor() {
+            @Override
+            public void process(final HttpRequest request, final HttpContext context) {
+                try {
+                    request.addHeader(new BasicHeader("X-Freenet-Insid", new MD5ChecksumCompute().compute(new MacUniqueIdService().getUUID()).hash));
```

Comment:
Can we atlast use SHA256 and truncate to 24 bits of it?[^1]

[^1]: https://en.wikipedia.org/wiki/MAC_address_anonymization#Ali_&_Dyo_Approach